src/features/checkOut/component/checkOut.tsx

```diff
@@ -1,268 +1,239 @@
+	const formatValues = (userTask: any) => {
+		const tasksArray = userTask.map((task: any) => {
+			return {
+				taskId: task.projectId,
+				taskName: task.taskCreate,
+				projectHours: '',
+				projectName: task.projectDetails.projectName,
+				id: task.id
+			};
+		});
+
+		return {
+			time: currentTime,
+			employees: [
+				[
+					{
+						task: '',
+						project: '',
+						projectHours: ''
+					}
+				]
+			],
+			tasks: userTask.length > 0 ? tasksArray : []
+		};
+	};
+
+	const selectedValidationSchema = () => {
+		if (isShowForm || enteredTask.length === 0) {
+			if (enteredTask.length > 0) {
+				return checkOutValidationWithOptSchema;
+			} else {
+				return checkOutwithNoTaskValidationSchema;
+			}
+		} else {
+			if (enteredTask.length > 0) {
+				return checkOutValidationSchema;
+			}
+		}
+	};
+
+	const validationSchema = selectedValidationSchema();
+
+	const form = useForm({
+		initialValues: formatValues(enteredTask),
+		validate: yupResolver(validationSchema)
+	});
+
+	const getProject = useCallback(() => {
+		setProjects(getProjectList(projectArray));
+	}, [projectArray]);
+
+	useEffect(() => {
+		getProject();
+	}, [getProject]);
+
+	const handleAddTaskBtn = () => {
+		setIsShowForm(!isShowForm);
+	};
+
+	useEffect(() => {
+		setOnChangeTime(currentTime);
+	}, []);
+
+	const handleTimeChange = (e) => {
+		const input = e.target.value;
+		let formattedTime = input
+			.replace(/\D/g, '')
+			.slice(0, 4)
+			.replace(/(\d{2})(\d{0,2})/, '$1:$2');
+
+		if (e.nativeEvent.inputType === 'deleteContentBackward') {
+			const lastChar = formattedTime.charAt(formattedTime.length - 1);
+			if (lastChar === ':') {
+				formattedTime = formattedTime.slice(0, -1);
+			}
+		}
+		setOnChangeTime(formattedTime);
+		form.setFieldValue('time', formattedTime);
+	};
+
+	/* API call for check out */
+	const handleCheckOut = useCallback(
+		async (values: any) => {
+			const project = values.employees.filter((value) => value.project && value.task && value.projectHours);
+
+			if (!project.length && !values.tasks.length) {
+				setIsConfirm(true);
+			} else {
+				const updatedTasks = values.tasks.map((item) => {
+					return {
+						taskId: item.id,
+						projectId: item.taskId,
+						taskName: item.taskName,
+						projectHours: item.projectHours
+					};
+				});
+
+				const updatedEmployees = values.employees.map((item) => {
+					return {
+						taskId: '',
+						projectId: item.project,
+						taskName: item.task,
+						projectHours: item.projectHours
+					};
+				});
+
+				let tasks;
+
+				if (enteredTask.length > 0) {
+					tasks = [...updatedTasks, ...updatedEmployees];
+				} else {
+					tasks = [...updatedEmployees];
+				}
+
+				const payload = {
+					outTime: values.time,
+					date: checkOutDate,
+					tasks: tasks.filter((value) => value.projectHours && value.projectId && value.taskName)
+				};
+				setIsLoading(true);
+				try {
+					await httpService.post(API_CONFIG.path.checkOut, payload).then((res: any) => {
```

Comment:
Remove response if you are not using it.

Reply:
done